app/controllers/PersonalDetailsController.scala

```diff
@@ -68,23 +71,24 @@ class PersonalDetailsController(applicationClient: ApplicationClient, userManage
       implicit val now: LocalDate = LocalDate.now
       GeneralDetailsForm.form.bindFromRequest.fold(
         errorForm => {
-          Future.successful(Ok(views.html.application.generalDetails(errorForm)))
+          Future.successful(Ok(views.html.application.generalDetails(GeneralDetailsForm.form.
+            bind(errorForm.data.cleanupFastPassFields()))))
         },
         gd => {
-          (for {
+          for {
             _ <- applicationClient.updateGeneralDetails(user.application.applicationId, user.user.userID,
-              removePostCodeWhenOutsideUK(gd), user.user.email)
+              removePostCodeWhenOutsideUK(gd).toExchange(user.user.email))
             _ <- userManagementClient.updateDetails(user.user.userID, gd.firstName, gd.lastName, Some(gd.preferredName))
             redirect <- updateProgress(data => data.copy(user = user.user.copy(firstName = gd.firstName, lastName = gd.lastName,
               preferredName = Some(gd.preferredName)), application = data.application.map(_.copy(applicationStatus = IN_PROGRESS))
             ))(_ => Redirect(routes.SchemePreferencesController.present()))
           } yield {
             redirect
-          })
+          }
         }
       )
   }
 
-  private def removePostCodeWhenOutsideUK(generalDetails: GeneralDetailsForm.Data): GeneralDetailsForm.Data =
+  private def removePostCodeWhenOutsideUK(generalDetails: GeneralDetailsForm.Data) =
     if (generalDetails.outsideUk.getOrElse(false)) generalDetails.copy(postCode = None) else generalDetails
```

Comment:
I have a similar problem because I do not clear up the form in javascript but I call the method sanitize, which I am not sure if it is right. However I think you can use a more generic name and then that name can be reused, so that we know it is the same problem in all controller we are dealing with.